syntax="proto2";
package leveldb;

option cc_generic_services = true;

message KVPair {
      required bytes key = 1;
      optional bytes value = 2;
}
message LeveldbRequest {
      required KVPair kv_pair = 2;
};

message LeveldbResponse {
      required bool success = 1;
      optional bytes value = 2;
      optional string error_message = 3;
};

service LeveldbService {
      rpc Execute(LeveldbRequest) returns (LeveldbResponse);
      rpc Put(LeveldbRequest) returns (LeveldbResponse);
      rpc Get(LeveldbRequest) returns (LeveldbResponse);
};
